cpp/src/io/copc_writer_internal.cpp

```diff
@@ -16,25 +16,28 @@ namespace copc::Internal
 
 size_t WriterInternal::OffsetToPointData() const
 {
-    size_t base_offset(375 + (54 + 160) + (54 + (34 + 4 * 6))); // header + COPC vlr + LAZ vlr (max 4 items)
+    size_t base_offset(las::LAZ_HEADER_SIZE + (54 + (34 + 4 * 6))); // header + LAZ vlr (max 4 items)
+    size_t eb_offset = config_->ExtraBytesVlr().size();
+    if (eb_offset > 0)
+        eb_offset += lazperf::vlr_header::Size;
+
+    size_t laz_offset = base_offset + eb_offset;
+
+    size_t copc_header_offset = (54 + 160); // COPC extra header size
```

Comment:
I think some of this could be refactored into a base class `OffetPointData` function. For example, the header size, laz vlr size, and extrabytes are common across all of them

Reply:
Only thing is that you can only call the base class statically, so I ended up doing a static function with a Config parameter, have a look and let me know if you like it.

Reply:
Ok I fixed all this code duplication with the dynamic pointer casts. Thanks for the guidance! It's beautiful :D